Marlin/src/module/probe.h

```diff
@@ -60,7 +60,7 @@
 
 #endif
 
-#if HAS_LEVELING && (HAS_BED_PROBE || ENABLED(PROBE_MANUALLY))
+#if HAS_BED_PROBE || ENABLED(PROBE_MANUALLY)
```

Comment:
`HAS_LEVELING` also included `ENABLED(MESH_BED_LEVELING)`.

`PROBE_SELECTED` might be a good choice to use. It expands to:
`#define PROBE_SELECTED (HAS_BED_PROBE || EITHER(PROBE_MANUALLY, MESH_BED_LEVELING))`

Reply:
I think `PROBE_SELECTED` leaves out probeless UBL, which was included with `HAS_LEVELING`. I believe that is ok, I think UBL only calls these functions if `HAS_BED_PROBE`.

If you delete the empty implementations in the else case Travis should tell you if anything will break.

Reply:
Correct me if i’m wrong, that HAS_BED_PROBE OR ENABLED(PROBE_MANUALLY) really covers all cases, where probe related functions are used. And these defines are self explanatory.

Reply:
I thought they would be used by `MESH_BED_LEVELING` (bedlevel.cpp), but it looks like they are not.

I now see they are used in `MARLIN_DEV_MODE` (M420.cpp). That feature will break without default implementations or rework, although the behavior with the defaults of all zeroes would be pretty useless.

Reply:
> I now see they are used in `MARLIN_DEV_MODE` (M420.cpp). That feature will break without default implementations or rework, although the behavior with the defaults of all zeroes would be pretty useless.

Just kidding. That is all inside a `HAS_LEVELING` check, so it should be ok in at least most cases.